diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index 1939fe84642b..8a23c3a3c3cd 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -94,9 +94,9 @@ deploy_libc_headers () {
 
 	# move asm headers to /usr/include/<libc-machine>/asm to match the structure
 	# used by Debian-based distros (to support multi-arch)
-	host_arch=$(dpkg-architecture -a$(cat debian/arch) -qDEB_HOST_MULTIARCH)
-	mkdir $pdir/usr/include/$host_arch
-	mv $pdir/usr/include/asm $pdir/usr/include/$host_arch/
+	#host_arch=$(dpkg-architecture -a$(cat debian/arch) -qDEB_HOST_MULTIARCH)
+	#mkdir $pdir/usr/include/$host_arch
+	#mv $pdir/usr/include/asm $pdir/usr/include/$host_arch/
 }
 
 version=$KERNELRELEASE
