From b4619984a579e5ba9431ea57afdc4ab244662a7a Mon Sep 17 00:00:00 2001
From: Merlijn Wajer <merlijn@wizzup.org>
Date: Mon, 23 Dec 2019 21:15:37 +0100
Subject: [PATCH 4/7] wip: more build hacks...

---
 scripts/package/builddeb | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index d24507eec234..64c47a0dc6ad 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -51,8 +51,8 @@ tmpdir="$objtree/debian/tmp"
 kernel_headers_dir="$objtree/debian/hdrtmp"
 libc_headers_dir="$objtree/debian/headertmp"
 dbg_dir="$objtree/debian/dbgtmp"
-packagename=linux-image-$version
-kernel_headers_packagename=linux-headers-$version
+packagename=linux-image-droid4
+kernel_headers_packagename=linux-headers-droid4
 libc_headers_packagename=linux-libc-dev
 dbg_packagename=$packagename-dbg
 
-- 
2.25.1

