diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index 09a2c558fcf6..ef5b29a77e1d 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -102,7 +102,7 @@ deploy_libc_headers () {
 version=$KERNELRELEASE
 tmpdir=debian/linux-image
 dbg_dir=debian/linux-image-dbg
-packagename=linux-image-$version
+packagename=linux-image-droid4
 dbg_packagename=$packagename-dbg
 
 if [ "$ARCH" = "um" ] ; then
