diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index 8a23c3a3c3cd..b3911e2d5a61 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -102,7 +102,7 @@ deploy_libc_headers () {
 version=$KERNELRELEASE
 tmpdir=debian/linux-image
 dbg_dir=debian/linux-image-dbg
-packagename=linux-image-$version
+packagename=linux-image-droid4
 dbg_packagename=$packagename-dbg
 
 if [ "$ARCH" = "um" ] ; then
