diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index b3911e2d5a61..e376ae46be08 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -205,10 +205,10 @@ sed -e "s,^\(exit 0\)$,\nln -snf /boot /boot/boot\n\1," -i "$tmpdir/DEBIAN/posti
 
 if [ "$ARCH" != "um" ]; then
 	deploy_kernel_headers debian/linux-headers
-	create_package linux-headers-$version debian/linux-headers
+	create_package linux-headers-droid4 debian/linux-headers
 
-	deploy_libc_headers debian/linux-libc-dev
-	create_package linux-libc-dev debian/linux-libc-dev
+	#deploy_libc_headers debian/linux-libc-dev
+	#create_package linux-libc-dev debian/linux-libc-dev
 fi
 
 create_package "$packagename" "$tmpdir"
