diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index ef5b29a77e1d..66620610d7f8 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -203,10 +203,10 @@ done
 
 if [ "$ARCH" != "um" ]; then
 	deploy_kernel_headers debian/linux-headers
-	create_package linux-headers-$version debian/linux-headers
+	create_package linux-headers-droid4 debian/linux-headers
 
-	deploy_libc_headers debian/linux-libc-dev
-	create_package linux-libc-dev debian/linux-libc-dev
+	#deploy_libc_headers debian/linux-libc-dev
+	#create_package linux-libc-dev debian/linux-libc-dev
 fi
 
 create_package "$packagename" "$tmpdir"
