diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index 66620610d7f8..c33a73a9c228 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -193,9 +193,23 @@ set -e
 export DEB_MAINT_PARAMS="\$*"
 
 # Tell initramfs builder whether it's wanted
-export INITRD=$(if_enabled_echo CONFIG_BLK_DEV_INITRD Yes No)
+export INITRD=No
 
 test -d $debhookdir/$script.d && run-parts --arg="$version" --arg="/$installed_image_path" $debhookdir/$script.d
+
+case "$script" in
+postinst)
+	depmod $version
+	rm -f /boot/boot /boot/zImage /boot/omap4-droid4-xt894.dtb
+	ln -s /boot/vmlinuz-$version /boot/zImage
+	ln -s /usr/lib/linux-image-droid4/omap4-droid4-xt894.dtb /boot/omap4-droid4-xt894.dtb
+	ln -s /boot /boot/boot
+	;;
+postrm)
+	rm -f /boot/zImage /boot/boot /boot/omap4-droid4-xt894.dtb
+	;;
+esac
+
 exit 0
 EOF
 	chmod 755 "$tmpdir/DEBIAN/$script"
