diff --git a/arch/arm/boot/dts/motorola-mapphone-common.dtsi b/arch/arm/boot/dts/motorola-mapphone-common.dtsi
index 8271746718fb..fa24762db04f 100644
--- a/arch/arm/boot/dts/motorola-mapphone-common.dtsi
+++ b/arch/arm/boot/dts/motorola-mapphone-common.dtsi
@@ -153,6 +153,39 @@ pwm9: dmtimer-pwm-9 {
 		ti,clock-source = <0x01>;
 	};
 
+	mapphone_touchscreen {
+			compatible = "touchscreen-buttons";
+			touchscreen_phandle = <&touchscreen>;
+			menu {
+				x-position = <0>;
+				y-position = <959>;
+				x-size = <256>;
+				y-size = <65>;
+				keycode = <KEY_F9>;
+			};
+			home {
+				x-position = <256>;
+				y-position = <959>;
+				x-size = <256>;
+				y-size = <65>;
+				keycode = <KEY_F10>;
+			};
+			back {
+				x-position = <512>;
+				y-position = <959>;
+				x-size = <256>;
+				y-size = <65>;
+				keycode = <KEY_F11>;
+			};
+			search {
+				x-position = <768>;
+				y-position = <959>;
+				x-size = <256>;
+				y-size = <65>;
+				keycode = <KEY_SEARCH>;
+			};
+	};
+
 	vibrator {
 		compatible = "pwm-vibrator";
 		pwms = <&pwm9 0 10000000 0>, <&pwm8 0 10000000 0>;
@@ -297,7 +330,7 @@ wlcore: wlcore@2 {
 };
 
 &i2c2 {
-	touchscreen@4a {
+	touchscreen: touchscreen@4a {
 		compatible = "atmel,maxtouch";
 		reg = <0x4a>;
 		pinctrl-names = "default";
diff --git a/arch/arm/configs/omap2plus_defconfig b/arch/arm/configs/omap2plus_defconfig
index e2422962fefa..83bee75cc010 100644
--- a/arch/arm/configs/omap2plus_defconfig
+++ b/arch/arm/configs/omap2plus_defconfig
@@ -251,6 +251,7 @@ CONFIG_TOUCHSCREEN_TSC2005=m
 CONFIG_TOUCHSCREEN_TSC2007=m
 CONFIG_INPUT_MISC=y
 CONFIG_INPUT_CPCAP_PWRBUTTON=m
+CONFIG_INPUT_TOUCHSCREEN_BUTTONS=m
 CONFIG_INPUT_TPS65218_PWRBUTTON=m
 CONFIG_INPUT_TWL4030_PWRBUTTON=m
 CONFIG_INPUT_UINPUT=m
